nixos/modules/services/networking/networkmanager.nix

```diff
@@ -229,15 +229,19 @@ in {
             source = mkOption {
               type = types.path;
               description = ''
-                A script.
+                Path to the hook script. Beware of the networkmanager dispatcher
+                service PATH (coreutils/iproute are available).
               '';
             };
 
             type = mkOption {
               type = types.enum (attrNames dispatcherTypesSubdirMap);
               default = "basic";
               description = ''
-                Dispatcher hook type. Only basic hooks are currently available.
+                Dispatcher hook type. Look up the hooks described at
```

Comment:
I am confused. `type` is defined as enum, which supports only 3 types (removed comment suggests that 2 of them don't work):
```
  dispatcherTypesSubdirMap = {
    "basic" = "";
    "pre-up" = "pre-up.d/";
    "pre-down" = "pre-down.d/";
  };
```
But suggested link lists 12 of them!

Reply:
I got confused at first too, my first tackle at the PR added:
 ```
dispatcherTypesSubdirMap = {
    "basic" = "";
    "pre-up" = "pre-up.d/";
    "up" = "";
    "pre-down" = "pre-down.d/";
    "down" = "";
    "vpn-pre-up" = "pre-up.d/";
    "vpn-up" = "";
    "vpn-pre-down" = "pre-down.d/";
    "vpn-down" = "";
    "hostname" = "";
    "dhcp4-change" = "";
    "dhcp6-change" = "";
    "connectivity-change" = "";
  };```
but then as I got to deploy my script realized that the addition might be misleading since the scripts end up in the same folder and they must check themselves their second parameter value ( e.g., if `[ "$2"= "dhcp4-change"] then .... fi` to act only for dchp4 events) so I reverted the change

Reply:
Ok, then maybe hide this option (make it internal, not deprecated)? I don't see any value in it except link in description.

Reply:
It has value because some events call hooks  only in the specific folders "pre-down.d/" /  "pre-up.d/";